### PR TITLE
Created LShape Function

### PR DESCRIPTION
**Pull request by mfsilva22**

### CODE REVIEW

File: frontend/require/PrairieDraw.js [hunk 1]
```diff
@@ -1687,7 +1687,76 @@ define(["sylvester", "sha1", "PrairieGeom"], function(Sylvester, Sha1, PrairieGe
...
     };
+	
+ /** Draw a rod with hinge points at start and end and the given width.
 
+        @param {Vector} startDw The first hinge point (center of circular end) in drawing coordinates.
+        @param {Vector} endDw The second hinge point (drawing coordinates).
+        @param {number} widthDw The width of the rod (drawing coordinates).
+    */
+    PrairieDraw.prototype.LshapeRod = function(startDw, centerDw, endDw, widthDw) {
+		//Create length vector
```
Comment (kevinwang) [inline frontend/require/PrairieDraw.js]: Convert all tab characters to four spaces.

File: frontend/require/PrairieDraw.js [hunk 2]
```diff
@@ -1687,7 +1687,76 @@ define(["sylvester", "sha1", "PrairieGeom"], function(Sylvester, Sha1, PrairieGe
...
+        var offsetLengthDw = endDw.subtract(startDw);
+		//Change width to vector form
+        var offsetWidthDw = offsetLengthDw.rotate(Math.PI/2, $V([0,0])).toUnitVector().x(widthDw);
+		
+			console.log(startDw);
```
Comment (kevinwang) [inline frontend/require/PrairieDraw.js]: Remove print statements.

File: frontend/require/PrairieDraw.js [hunk 3]
```diff
@@ -1687,7 +1687,76 @@ define(["sylvester", "sha1", "PrairieGeom"], function(Sylvester, Sha1, PrairieGe
...
+        this._ctx.beginPath();
+        this._ctx.moveTo(0, rPx);
+				console.log(startPx);
+				console.log(endPx);
+				console.log(rPx);
+				console.log(lengthPx);
+				
+		//quadraticCurveTo(cp1x, cp1y, x, y)
+		//Draws a quadratic Bézier curve from the current pen position to the end point specified by x and y, using the control point specified by cp1x and cp1y.
+		//this._ctx.quadraticCurveTo(centerDw.e(1),centerDw.e(2), startPx.e(1),startPx.e(2));
```
Comment (kevinwang) [inline frontend/require/PrairieDraw.js]: Remove commented code.

**Comment by kevinwang** [comment]
Can you include screenshots of the feature?

**Comment by mfsilva22** [comment]
Hi Matt
Should we delete this pull request so that people disregard it for the time being? Can you explain the Kevin what is going on?
Thanks
Mariana

Sent from my iPhone

On Jul 19, 2016, at 9:51 PM, Kevin Wang <notifications@github.com<mailto:notifications@github.com>> wrote:


Can you include screenshots of the feature?

**Comment by mwest1066** [comment]
Mariana, I'll leave the PR open for now, because I guess you are actually going to implement this feature? There isn't any problem to having it open and letting people comment, especially when some of the comments are entirely helpful (e.g., using spaces rather than tabs, can you fix this in your editor?).

Kevin, thanks for the comments! This is a test PR that I made with Mariana for her to understand how github and the PR process works. The current new code is just a copy of an existing function and it now needs to be modified to actually implement the feature. I think Mariana's plan is to do this next, so we should be seeing more code updates soon!

**Comment by mfsilva22** [comment]
The comments are great. Thanks! I will "clean up" the function once we are done with it. Stephanie is the one writing all the comments and I think she needs them at this time. Because we are working from 3 different local machines, I am anticipating a lot of commits in the next few days. I can post a message here once I feel someone can take a look at it. Would that work?

**Comment by mfsilva22** [comment]
I completed my first draft of the function LShapeRod. I just noticed that I still have tabs instead of the 4 spaces. I will open the file using another editor to fix this problem soon (I don't see the difference when opening the file using Notepad ++). See below the screenshot of the drawing for two different set of coordinates. Please let me know what you think. Thanks!
lbracket

**Comment by rbessick5** [comment]
Notepad++ can take care of the TAB to space problem for you automatically, if you prefer (I had to do this for my own code not too long ago).

You first have to set the TAB width: open Settings/Preferences, select "Tab settings" (about halfway down the left-hand list), then make sure the Tab size is 4 and "Replace with space" is checked. This also means any time you press the TAB key, Notepad++ will input 4 spaces instead.

To convert any existing TAB characters, highlight the whole document, then select Edit/Blank Operations/TAB to space. That will replace any TAB characters in the highlighted area with 4 spaces automatically.

**Comment by mwest1066** [comment]
Closing this issue because we merged the code into fa55f5358a7de420ef87f1dea42c117f83e1ff9b

(it didn't autoclose because of the remote-tracking merge, which I didn't include)

**Pull request closed without merging**